src/seedu/addressbook/data/person/Address.java

```diff
@@ -36,21 +142,50 @@ public static boolean isValidAddress(String test) {
         return test.matches(ADDRESS_VALIDATION_REGEX);
     }
 
+    /**
+     * 
+     * @param input
+     *            string to take the a/ out from
+     * @return return the string without the a/ prefix
+     */
+    private static String removeAddressPrefix(String input) {
+        if (input.startsWith("a/")) {
+            return input.replaceFirst("a/", "");
+        } else {
+            return input;
+        }
+    }
+
     @Override
     public String toString() {
-        return value;
+        return this.block.getBlockNumber() + SPACE + this.street.getStreetName() + SPACE 
+               + this.unit.getUnitNumber() + SPACE + this.postalCode.getPostalCode();
     }
 
     @Override
     public boolean equals(Object other) {
-        return other == this // short circuit if same object
-                || (other instanceof Address // instanceof handles nulls
-                && this.value.equals(((Address) other).value)); // state check
+        if (other == this) {
+            return true; // short circuit if same object
+        } else if (other instanceof Address) {// instanceof handles nulls
+            Address otherAddress = (Address) other;
+            boolean isBlockEqual = this.block.getBlockNumber() == otherAddress.block.getBlockNumber();
+            boolean isStreetEqual = this.street.getStreetName().equals(otherAddress.street.getStreetName());
+            boolean isUnitEqual = this.unit.getUnitNumber().equals(otherAddress.unit.getUnitNumber());
+            boolean isPostalEqual = this.postalCode.getPostalCode() == otherAddress.postalCode.getPostalCode();
+            return isBlockEqual && isStreetEqual && isUnitEqual && isPostalEqual;
+        } else {
+            return false;
+        }
     }
 
     @Override
     public int hashCode() {
-        return value.hashCode();
+        int code = LARGE_PRIME;
```

Comment:
there is no need to use a large prime number. you can use XOR on the components' hashCodes